### PR TITLE
refactor: extract Facade to its own class

### PR DESCRIPTION
**Pull request by sdelamo**

### CODE REVIEW

File: crac/src/main/java/io/micronaut/crac/support/CracFacadeResource.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        orderedCracResource.beforeCheckpoint(createCracContext(context));
+    }
+
+    @Override
+    public void afterRestore(Context<? extends Resource> context) throws Exception {
+        orderedCracResource.afterRestore(createCracContext(context));
+    }
+
+    private CracContext createCracContext(Context<? extends Resource> context) {
+        return new DefaultCracContext(() -> (Context<Resource>) context);
```
Comment (sdelamo) [inline crac/src/main/java/io/micronaut/crac/support/CracFacadeResource.java:47]: It passes the `context` passed  to original methods: `org.crac.Resource::beforeCheckpoint` and `org.crac.Resource::afterRestore`. 

I think the cast to `Context<Resource>` is correct. Not 100% sure.

File: crac/src/main/java/io/micronaut/crac/support/CracFacadeResource.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import io.micronaut.core.annotation.Internal;
+import org.crac.Context;
+import org.crac.Resource;
+
+/**
+ * The bridge between a Micronaut CRaC Resource and the CRaC api.
+ * @author Tim Yates
+ * @since 3.7.0
+ */
+@Internal
```
Comment (timyates) [inline crac/src/main/java/io/micronaut/crac/support/CracFacadeResource.java:27]: Does this also need `@Experimental`?

File: crac/src/main/java/io/micronaut/crac/support/CracFacadeResource.java (hunk 1 above)
Comment (timyates) [reply crac/src/main/java/io/micronaut/crac/support/CracFacadeResource.java:47]: I spent a long time in this kind of generic hell on Thurs/Friday...  I think this is ok though 👍

**Review by timyates — APPROVED** [review]

File: crac/src/main/java/io/micronaut/crac/support/CracFacadeResource.java (hunk 2 above)
Comment (sdelamo) [reply crac/src/main/java/io/micronaut/crac/support/CracFacadeResource.java:27]: As it is `@Internal`, I think it is ok. But will annotate everything `@Experimental` nonetheless.

**Pull request merged**